org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AbstractRunnableDeviceDelegate.java

```diff
@@ -0,0 +1,183 @@
+ * with GDA. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.eclipse.scanning.sequencer;
+
+import org.eclipse.dawnsci.nexus.NXdetector;
+import org.eclipse.dawnsci.nexus.NexusException;
+import org.eclipse.dawnsci.nexus.NexusScanInfo;
+import org.eclipse.dawnsci.nexus.builder.NexusObjectProvider;
+import org.eclipse.dawnsci.nexus.builder.NexusObjectWrapper;
+import org.eclipse.scanning.api.annotation.scan.WriteComplete;
+import org.eclipse.scanning.api.event.core.IPublisher;
+import org.eclipse.scanning.api.event.scan.ScanBean;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.LevelInformation;
+import org.eclipse.scanning.api.scan.ScanInformation;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is the abstract base class of all classes which define the delegated functionality of an
+ * AreaDetectorRunnableDeviceProxy object.
+ * <P>
+ * Since this defines all functions through inheritance rather than through annotations, and it doesn't depend on
+ * interface default methods, it can be used as a base class for delegates defined in Jython.
+ * <P>
+ * Note that an object of this class must be initialised with a reference to the object it is going to provide
+ * delegated functionality to and setDelegate() on that object should then be immediately called with this one.
+ * <p>
+ * Example of use:
+ * </P>
+ * <pre>{@code
+areaDetectorRunnableDeviceProxyFinder = finder.find("areaDetectorRunnableDeviceProxyFinder")
+areaDetectorRunnableDeviceProxy = areaDetectorRunnableDeviceProxyFinder.getRunnableDevice()
+
+from jythonAreaDetectorRunnableDeviceDelegate import JythonAreaDetectorRunnableDeviceDelegate
+jythonAreaDetectorRunnableDeviceDelegate = JythonAreaDetectorRunnableDeviceDelegate(areaDetectorRunnableDeviceProxy)
+areaDetectorRunnableDeviceProxy.setDelegate(jythonAreaDetectorRunnableDeviceDelegate)
+areaDetectorRunnableDeviceProxy.register()
+ * }</pre>
+ */
+public abstract class AbstractRunnableDeviceDelegate {
+
+	private static final Logger logger = LoggerFactory.getLogger(AbstractRunnableDeviceDelegate.class);
+
+	RunnableDeviceProxy runnableDeviceProxy;
+
+	public AbstractRunnableDeviceDelegate(RunnableDeviceProxy runnableDeviceProxy) {
+		this.runnableDeviceProxy = runnableDeviceProxy;
+	}
+
+	public RunnableDeviceProxy getRunnableDeviceProxy() {
+		return runnableDeviceProxy;
+	}
+
+	// Delegated AbstractRunnableDevice<AreaDetectorRunnableDeviceModel> methods
+
+	@SuppressWarnings("unused")
+	public void configure(Object model) throws ScanningException {
+		logger.trace("configure({}) on {}", model, runnableDeviceProxy.getName());
+	}
+
+	// Delegated interface IRunnableDevice<AreaDetectorRunnableDeviceModel> methods
+
+	@SuppressWarnings("unused")
+	public void run(IPosition position) throws ScanningException, InterruptedException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.eclipse.scanning.api.scan.ScanningException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.InterruptedException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)